### PR TITLE
Enhance documentation

### PR DESCRIPTION
**Pull request by nhoizey**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,16 +2,54 @@
...
-- Step 1: get actual contexts (viewports and screen densities) of site visitors
-- Step 2: get variations of image width across viewport widths
-- Step 3: compute optimal n widths from both datasets
+## How does it work?
 
-See details in [the full documentation](https://cleverage.github.io/responsive-image-widths/)
+It takes 3 steps for `responsive-image-widths` to find the best widths to put in the `srcset` attribute of a responsive image:
+
+- take Real User Monitoring (RUM) of viewport widths and screen densities
+- list the image's widths across all of these viewport widths
```
Comment (ryuran) [inline README.md]:
```suggestion
- list the image’s widths across all of these viewport widths
```

File: README.md [hunk 2]
```diff
@@ -2,16 +2,54 @@
...
+- compute the optimal image widths to put in the `srcset` attribute to cover all these needs
+
+Learn more in [the full documentation](https://cleverage.github.io/responsive-image-widths/).
+
+## Usage
+
+To install and run this application, you'll need [Node.js](https://nodejs.org/en/download/) (which comes with [npm](http://npmjs.com)) installed on your computer.
```
Comment (ryuran) [inline README.md]:
```suggestion
To install and run this application, you’ll need [Node.js](https://nodejs.org/en/download/) (which comes with [npm](http://npmjs.com)) installed on your computer.
```

File: docs/art-direction.md [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+<picture>
+  <source media="(min-width: 800px)" srcset="…" sizes="…">
+  <img srcset="…" sizes="…" alt="…">
+</picture>
+```
+
+You will have to run the script twice, with (at least) these parameters, to get widths for both `srcset`s:
+
+```shell
+npx responsive-image-widths --max-viewport 799
```
Comment (ryuran) [inline docs/art-direction.md]: this commands can’t currently work without `--verbose` or `--destFile`

File: docs/index.md [hunk 4]
```diff
@@ -1,31 +1,116 @@
...
-- [Step 1: get actual contexts (viewports and screen densities) of site visitors](/responsive-image-widths/step1.html)
-- [Step 2: get variations of image width across viewport widths](/responsive-image-widths/step2.html)
-- [Step 3: compute optimal n widths from both datasets](/responsive-image-widths/step3.html)
+Using responsive images in every projects, we wanted to be able to make it as efficient as possible. The main difficulty is choosing the image widths we put in `srcset` attributes, because nothing in the standard tells us about this.
 
-## How to deal with multiple `<source>` with `mix/max-width` media queries (Art Direction)
+## How can we choose the responsive image widths?
 
-If you have some code like this:
+We didn't invent anything here, we're standing on the shoulders of giants.
```
Comment (ryuran) [inline docs/index.md]:
```suggestion
We didn’t invent anything here, we're standing on the shoulders of giants.
```

File: docs/index.md [hunk 5]
```diff
@@ -1,31 +1,116 @@
...
-  <source media="(min-width: 800px)" srcset="…" sizes="…">
-  <img srcset="…" sizes="…" alt="…">
-</picture>
+Back in 2015, [Jason Grigsby](https://cloudfour.com/is/jason-grigsby/) wrote this in [Responsive Images 101, Part 9: Image Breakpoints](https://cloudfour.com/thinks/responsive-images-101-part-9-image-breakpoints/):
+
+> we want to provide multiple image sources because of performance concerns, different screen densities, etc. [but] we can’t simply reuse our responsive layout breakpoints for our images.
+
+Jason presented a few ways to decide which image sizes to put in the `srcset` attribute of responsive images.
+
+[Cloudinary](https://cloudinary.com) then developed the [Responsive Image Breakpoints Generator](https://www.responsivebreakpoints.com/), based on Jason's [setting image breakpoints based on a performance budget](https://cloudfour.com/thinks/responsive-images-101-part-9-image-breakpoints/#setting-image-breakpoints-based-on-a-performance-budget) idea. It was already a good optimization.
```
Comment (ryuran) [inline docs/index.md]:
```suggestion
[Cloudinary](https://cloudinary.com) then developed the [Responsive Image Breakpoints Generator](https://www.responsivebreakpoints.com/), based on Jason’s [setting image breakpoints based on a performance budget](https://cloudfour.com/thinks/responsive-images-101-part-9-image-breakpoints/#setting-image-breakpoints-based-on-a-performance-budget) idea. It was already a good optimization.
```

File: docs/index.md [hunk 6]
```diff
@@ -1,31 +1,116 @@
...
+
+> We’d start by defining a budget for the amount of wasted bytes that the browser would be allowed to download above what is needed to fit the size of the image in the page.
+
+But we believe the most efficient of Jason's ideas is [setting image breakpoints based on most frequent requests](https://cloudfour.com/thinks/responsive-images-101-part-9-image-breakpoints/#setting-image-breakpoints-based-on-most-frequent-requests), inspired by a discussion with [Yoav Weiss](https://blog.yoav.ws/) from Akamai (who made [Blink and webkit support responsive images](https://blog.yoav.ws/by_the_people/) before joining Akamai) and [Ilya Grigorik](https://www.igvita.com/) from Google:
```
Comment (ryuran) [inline docs/index.md]:
```suggestion
But we believe the most efficient of Jason’s ideas is [setting image breakpoints based on most frequent requests](https://cloudfour.com/thinks/responsive-images-101-part-9-image-breakpoints/#setting-image-breakpoints-based-on-most-frequent-requests), inspired by a discussion with [Yoav Weiss](https://blog.yoav.ws/) from Akamai (who made [Blink and webkit support responsive images](https://blog.yoav.ws/by_the_people/) before joining Akamai) and [Ilya Grigorik](https://www.igvita.com/) from Google:
```

File: docs/index.md [hunk 7]
```diff
@@ -1,31 +1,116 @@
...
+
+> For these organizations, they can tie their image processing and breakpoints logic to their analytics and change the size of the images over time if they find that new image sizes are getting requested more frequently.
+
+Jason Grigsby also wrote in the same article that [humans shouldn’t be doing this](https://cloudfour.com/thinks/responsive-images-101-part-9-image-breakpoints/#humans-shouldnt-be-doing-this), and we agree. That's why we starting developing `responsive-image-widths`.
```
Comment (ryuran) [inline docs/index.md]:
```suggestion
Jason Grigsby also wrote in the same article that [humans shouldn’t be doing this](https://cloudfour.com/thinks/responsive-images-101-part-9-image-breakpoints/#humans-shouldnt-be-doing-this), and we agree. That’s why we starting developing `responsive-image-widths`.
```

File: docs/index.md [hunk 8]
```diff
@@ -1,31 +1,116 @@
...
+
+It takes 3 steps for `responsive-image-widths` to find the best widths to put in the `srcset` attribute of a responsive image:
+
+- [Step 1: take Real User Monitoring (RUM) of viewport widths and screen densities used on the website](/responsive-image-widths/step1.html)
+- [Step 2: detect the image's widths across all of these viewport widths](/responsive-image-widths/step2.html)
+- [Step 3: compute the optimal image widths to put in the `srcset` attribute to cover all these needs](/responsive-image-widths/step3.html)
+
+## Getting started
+
+To install and run this application, you'll need [Node.js](https://nodejs.org/en/download/) (which comes with [npm](http://npmjs.com)) installed on your computer.
```
Comment (ryuran) [inline docs/index.md]:
```suggestion
To install and run this application, you’ll need [Node.js](https://nodejs.org/en/download/) (which comes with [npm](http://npmjs.com)) installed on your computer.
```

File: docs/index.md [hunk 9]
```diff
@@ -1,31 +1,116 @@
...
+Examples:
+  npx cli.js --contextsFile ./contexts.csv --url 'https://example.com/' --selector 'main img[srcset]:first-of-type' --verbose
+  npx cli.js -c ./contexts.csv -u 'https://example.com/' -s 'main img[srcset]:first-of-type' -i 320 -x 1280 -a ./variations.csv -f ./srcset-widths.txt -v
+```
+
+See [details about each option](/responsive-image-widths/options.html).
+
+## Examples and use cases
+
+There are a few examples in the project's repository: [examples](https://github.com/cleverage/responsive-image-widths/tree/master/examples)
```
Comment (ryuran) [inline docs/index.md]:
```suggestion
There are a few examples in the project’s repository: [examples](https://github.com/cleverage/responsive-image-widths/tree/master/examples)
```

File: docs/art-direction.md (hunk 3 above)
Comment (nhoizey) [reply docs/art-direction.md]: Indeed…

**Pull request merged**